pkgs/unified_analytics/lib/src/enums.dart

```diff
@@ -153,6 +153,16 @@ enum DashEvent {
     required this.description,
     this.toolOwner,
   });
+
+  /// This takes in the string label for a given [DashEvent] and returns the
+  /// enum for that string label.
+  static DashEvent getDashEventByLabel(String label) {
+    for (final event in DashEvent.values) {
+      if (event.label == label) return event;
+    }
+
+    throw Exception('The event $label is not a valid DashEvent enum value');
```

Comment:
do we handle this?

Reply:
Yes, this gets called from `Event._fromJson` which gets called in the static method `Event.fromJson`. The static method has a `Event?` return type so it will return null if we encounter this `Exception`

Reply:
Ahh, I see. Instead of throwing, can we have all of these intermediate functions be nullable, and then always check if the return value was null, and if so immediately return null.

I'd rather avoid this sentinel exception because:

1. If, in the future someone adds call site to `DashEvent.getDashEventByLabel()` they would have to notice that it might throw--if the return type is nullable, the compiler will force the caller to handle the potential null value
2. By catching the generic `Exception` several levels up, we may unintentionally end up swallowing unexpected exceptions, when we may want to know something went wrong.

Reply:
That's a good point, i have swapped out the thrown exception for a nullable return type on `DashEvent.getDashEventByLabel`.

Would this have been better if I had thrown something other than an `Exception`? Perhaps something more specific to this package?